validation-test/stdlib/FixedPointConversion.swift.gyb

```diff
@@ -97,6 +124,105 @@ FixedPointConversionFailure.test("${Other}To${Self}Conversion/dest=${testValue}"
 %     end # for testValue in ...
 %   end # for in all_integer_types (Other)
 % end # for in all_integer_types (Self)
+
+%   for Other, otherSignificandBits in floatNameToSignificandBits.iteritems():
+%     otherMin = intMin(otherSignificandBits, False)
+%     otherMax = intMax(otherSignificandBits, False)
+
+%     if Other == 'Float80':
+#if arch(i386) || arch(x86_64)
+%     end
+
+%     for testValue in [repr(value) for value in [selfMin, selfMax, selfMin - 0.1, selfMax + 0.1, otherMin, otherMax, 0.0, -0.0, 0.1, -0.1]]:
+        
+%       if testValue < otherMin or testValue > otherMax: 
+%          # Can't construct `other` value, do nothing and continue.
+
+%       elif testValue >= selfMin and testValue <= selfMax and testValue % 1 == 0:
+
+FloatingPointConversionTruncations.test("${Other}To${Self}Conversion/dest=${testValue}") {
+  let input = get${Other}(${testValue})
+  let result = ${Self}(input)
+  var resultConvertedBack = ${Other}(result)
+  expectEqual(${testValue}, resultConvertedBack)
+}
+
+FloatingPointConversionFailure.test("${Other}To${Self}FailableConversion/dest=${testValue}") {
+  let input = get${Other}(${testValue})
+  var result = ${Self}(exactly: input)
+  expectNotEqual(result, nil)
+}
+
+%       else:
+
+%         if testValue > selfMax:
+FloatingPointConversionTraps.test("${Other}To${Self}Conversion/dest=${testValue}").crashOutputMatches(getTooLargeMessage()).code {
+expectCrashLater()
+%         elif testValue < selfMin:
+FloatingPointConversionTraps.test("${Other}To${Self}Conversion/dest=${testValue}").crashOutputMatches(getTooSmallMessage()).code {
+expectCrashLater()
+%         else:
+FloatingPointConversionTruncations.test("${Other}To${Self}Conversion/dest=${testValue}") {
+%         end
+  let input = get${Other}(${testValue})
+  var result = ${Self}(input)
+  var resultConvertedBack = ${Other}(result)
+  expectNotEqual(input, resultConvertedBack)
+}
+
+FloatingPointConversionFailure.test("${Other}To${Self}Conversion/dest=${testValue}") {
+  let input = get${Other}(${testValue})
+  var result = ${Self}(exactly: input)
+  expectEqual(nil, result)
+}
+%       end
+
+%     end # for in testValues
+
+// Test Always-Trapping conversions.
+
+FloatingPointConversionTraps.test("${Self}/${Other}/+inf")
+  .crashOutputMatches(getInfiniteOrNaNMessage()).code {
+  expectCrashLater()
+  _blackHole(${Self}(get${Other}(+${Other}.infinity)))
+}
+
+%     if selfSigned:
```

Comment:
I think this condition is not needed.  `UInt(-Float.infinity)` and we should test that.